blog/_posts/2017-09-06-gsoc-derivative_operators.md

```diff
@@ -0,0 +1,156 @@
+
+        julia> using DiffEqOperators, DifferentialEquations, Plots
+        julia> x = collect(-pi : 2pi/511 : pi);
+        julia> u0 = -(x - 0.5).^2 + 1/12;
+        julia> A = DerivativeOperator{Float64}(2,2,2pi/511,512,:Dirichlet,:Dirichlet;BC=(u0[1],u0[end]));
+
+Now solving equation as an ODE we have:-
+    
+        julia> prob1 = ODEProblem(A, u0, (0.,10.));
+        julia> sol1 = solve(prob1, dense=false, tstops=0:0.01:10);
+        # try to plot the solution at different time points using
+        julia> plot(x, [sol1(i) for i in 0:1:10])
+
+![Heat Equation](/images/blog/2017-09-06-gsoc-derivative_operators/heat_eqn_D1.png)
+
+
+## Where do central derivatives fail?
+
+Not all PDEs can be solved with central derivatives, for example the **KdV wave equation**. After a few iterations of the ODE solver the wave begins to split ie. it becomes unstable very quickly.
+
+![KdV using central derivatives](/images/blog/2017-09-06-gsoc-derivative_operators/kdv_derivative.png)
+
+
+For these very cases the [upwind scheme](https://en.wikipedia.org/wiki/Upwind_scheme) has been devised. It denote a class of numerical discretization methods for solving hyperbolic PDEs. They attempt to discretize hyperbolic PDEs by using differencing biased in the direction determined by the sign of the characteristic speeds. For example the 1D linear advection equation
+$$
+\frac{\partial u}{\partial t} + a\frac{\partial u}{\partial x}=0
+$$
+describes a wave propagating along the x-axis with a velocity $a$. If $a$ is positive, the traveling wave solution of the equation above propagates towards the right, the left side is then called the upwind side and the right side is called the the downwind side. If the finite difference scheme for the spatial derivative, $\frac{\partial u}{\partial x}$ contains more points in the upwind side, the scheme is called **upwind scheme**. Considering a case of 2nd upwind scheme, define
+
+$$ a^{+} = max(a,0)$$
+
+$$a^{-} = min(a,0)$$
+
+$$u_x^- = \frac{3u_i^n-4u_{i-1}^n+u_{i-2}^n}{2\Delta x}$$
+
+$$u_x^+ = \frac{-u_{i+2}^n+4u_{i+1}^n-3u_{i}^n}{2\Delta x} $$
+
+The general solution can then be written as follows:-
+$u_i^{n+1} = u_i^n - \Delta t[a^{+}u_x^{-} + a^{-}u_x^{+}]$
+
+The solution of the **KdV equation** using upwind operator looks better.
+
+![KdV solved using Upwind operators](/images/blog/2017-09-06-gsoc-derivative_operators/kdv_upwind.png)
+
+## Future Work
+Although vanilla `DerivativeOperators` and the `UpwindOperators` form the major part of DiffEqOperators there is still a lot to be done. A major functionality which is half implemented is application of DiffEqOperators on high dimensional spaces. Currently we support mixed and normal derivatives on 2D spaces only. There are open [issues](https://github.com/JuliaDiffEq/DiffEqOperators.jl/issues/20) and implementation [ideas](https://github.com/JuliaDiffEq/DiffEqOperators.jl/issues/21) on the issues page.
```

Comment:
You should probably mention current issues: Robin BCs.

Reply:
yeah maybe that will be more proper.

Reply:
Cool. This done.